mktestimg.c

```diff
@@ -22,87 +22,123 @@ enum {
     FORMAT_COUNT,
 };
 
-char *copystr(const char *string) {
-    long len = strlen(string); 
-    char *out = malloc(sizeof(char) * (len + 1));
-    strcpy(out, string); 
-    out[len] = '\0';
+char* copy_str(const char* string) {
+    if (string == NULL) {
+        fprintf(stderr, "error: The string cannot be empty.\n");
+        exit(EXIT_FAILURE);
+    }
+
+    long len = strlen(string) + 1;
+    char* out = malloc(len);
+
+    strcpy(out, string);
+    out[strcspn(out, "\n")] = 0;
```

Comment:
This is very very unsafe and redundant. Consider the following example:
``./mktestimg -w 100 -h 100 -c 4 -f 1 -o "out ``
``.png"``
Here, the name of the output file is scattered through multiple lines. This should be possible.
Your PR causes the ".png" part to be omitted while the master deals with this correctly. The old solution is completely correct.

Reply:
Which line redundant here? The pointer passed as the parameter must be checked if it's `NULL` – otherwise, do you really think `copy_str()` would work if the string is empty?

Reply:
Yes, `out[strcspn(out, "\n")]` will of course stop at the first occurrence. Well, I tried to define an alternative approach to the problem, but I was unaware if the argument could contain a *newline* that certainly eliminates the next characters.